libcontainerd/container_windows.go

```diff
@@ -249,7 +249,13 @@ func (ctr *container) waitExit(process *process, isFirstProcessToStart bool) err
 						}
 						logrus.Error(err)
 					} else {
-						ctr.client.Create(ctr.containerID, ctr.ociSpec, ctr.options...)
+						if err := ctr.client.Create(ctr.containerID, ctr.ociSpec, ctr.options...); err != nil {
+							si.State = StateExit
```

Comment:
Indeed, not introduced by this pr but doesn't look ideal. Shouldn't cause problems atm because `<-wait` always blocks if `restart` is set.  I think this goroutine should be after `StateChanged()` call with `if si.State == StateRestart` condition. Same problem in linux version. @mlaventure wdyt?